From 942656546cd307829e0359d9a89fb6f012a155de Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Fri, 24 Jan 2014 19:43:05 +0000
Subject: [PATCH] builder: Add documentation for enabling Puppet (agent) in a
 guest.

(cherry picked from commit 080300dfec1dd603da2f7d53e72dd3e056317fb8)
---
 builder/virt-builder.pod | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/builder/virt-builder.pod b/builder/virt-builder.pod
index 52bf536..ed08f3d 100644
--- a/builder/virt-builder.pod
+++ b/builder/virt-builder.pod
@@ -1121,6 +1121,29 @@ following could be used to boot the virtual machine:
 As with libvirt, it is very important that the correct format is
 chosen.  It will be C<raw> unless the I<--format> option was used.
 
+=head2 CONFIGURATION MANAGEMENT
+
+=head3 Puppet
+
+To enable the Puppet agent in a guest, install the package, point
+the configuration at your Puppetmaster, and ensure the agent runs
+at boot.
+
+A typical virt-builder command would be:
+
+ virt-builder fedora-20 \
+   --hostname client.example.com \
+   --update \
+   --install puppet \
+   --edit '/etc/puppet/puppet.conf:
+       s/^\[agent\]/[agent]\n  server = puppetmaster.example.com/' \
+   --run-command 'systemctl enable puppet' \
+   --selinux-relabel
+
+The precise instructions vary according to the Linux distro.  For
+further information see:
+L<http://docs.puppetlabs.com/guides/installation.htm>
+
 =head2 DEBUGGING BUILDS
 
 If virt-builder itself fails, then enable debugging (I<-v>) and report
-- 
1.8.5.3

